### PR TITLE
EREGCSC-2267 Sanitize filenames

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves # EREGCSC-2267 Sanitize file names

**Description-**

We want to make sure filenames dont have weird characters in it.  Before we only did commas but now we are just adding a slew of them
**This pull request changes...**

- Use djangos slugify to clean out characters. https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.text.slugify

**Steps to manually verify this change...**

1. Log into the admin panel.
2. Upload a file with a bunch of weird characters. Quotes, underscores, hyphens etc.
3. Characters will be cleaned out after saving.  Should be mostly alphanumeric.

### CODE REVIEW

**Review by peggles2 — CHANGES_REQUESTED** [review]
Nice work. Can you please write a simple test for this as well?

**Comment by thwalker6** [comment]
> Nice work. Can you please write a simple test for this as well?

Slugify if a django library which would already have its own test.  I wouldnt write a unit test for it.

File: solution/backend/file_manager/admin.py [hunk 1]
```diff
@@ -54,16 +55,11 @@ class UploadedFileAdmin(BaseAdmin):
...
-    def clean_file_name(self, name):
-        name = name.replace(',', '')
-        return name
-
     def save_model(self, request, obj, form, change):
         path = form.cleaned_data.get("file_path")
         if path:
-            obj.file_name = self.clean_file_name(path._name)
+            file_name, extension = path._name.split('.')
+            obj.file_name = f"{slugify(file_name).replace('-', ' ')}.{extension}"
```
Comment (peggles2) [inline solution/backend/file_manager/admin.py:62]: Although you use slugify you are also replacing the '-' with ' ' so a simple test would be good to have here that we expect the file names to be cleaned up in this way. Thanks

**Review by peggles2 — CHANGES_REQUESTED** [review]

File: solution/backend/file_manager/admin.py (hunk 1 above)
Comment (thwalker6) [reply solution/backend/file_manager/admin.py]: removed replace to simplify the code.

**Pull request merged**